examples/tutorials/advanced/focal_mechanisms.py

```diff
@@ -0,0 +1,351 @@
+"""
+Plotting focal mechanisms
+=========================
+
+Focal mechanisms can be plotted as beachballs with the :meth:`pygmt.Figure.meca`
+method.
+
+The focal mechanism data or parameters can be provided as different input types:
+an ASCII file, 1-D or 2-D :class:`numpy.array`, dictionary, :class:`pandas.Dataframe`.
+Different conventions to define the focal mechanism are supported: Aki and Richards
+(``"aki"``), global CMT (``"gcmt"``), moment tensor (``"mt"``), partial focal
+mechanism (``"partial"``), and, principal axis (``"principal_axis"``). Please refer to
+the documentation of :meth:`pygmt.Figure.meca` regarding how to set up the input
+data in respect to the chosen input type and convention (i.e., the expected column
+order, keys, or column names).
+
+This tutorial focus on how to adjust the display of the beachballs:
+
+- Filling the quadrants
+- Plotting the components of a seismic moment tensor
+- Adjusting the outlines
+- Highlighting the nodal planes
+- Adding offset from the event location
+- Adding a label
+- Using size-coding and color-coding
+"""
+
+# %%
+import pandas as pd
+import pygmt
+
+# Set up arguments for basemap
+region = [-5, 5, -5, 5]
+projection = "X10c/4c"
+frame = ["af", "+ggray90"]
+
+
+# %%
+# Set up the focal mechanism data
+# -------------------------------
+#
+# Store focal mechanism parameters for one event in a dictionary following the
+
+# moment tensor convention
+mt_dict_single = {
+    "mrr": 4.71,
+    "mtt": 0.0381,
+    "mff": -4.74,
+    "mrt": 0.399,
+    "mrf": -0.805,
+    "mtf": -1.23,
+    "exponent": 24,
+}
+# Aki and Richards convention
+aki_dict_single = {"strike": 318, "dip": 89, "rake": -179, "magnitude": 7.75}
+
+
+# %%
+# Plotting a single beachball
+# ---------------------------
+#
+# Required parameters are ``spec` and ``scale`` as well as ``longitude``, ``latitude``
```

Comment:
Need to mention that `scale` controls the size of beachballs.

Reply:
Please see commit https://github.com/GenericMappingTools/pygmt/pull/2550/commits/350d01d2cf2066d79d7f7651a99252099df515ff for explaining `scale`.